platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookValidator.java

```diff
@@ -92,6 +94,66 @@ public static boolean validNextId(final AddressBook previousAddressBook, final A
         return true;
     }
 
+    /**
+     * Validate that the new next node id is greater than or equal to the old next node id, and that all nodes that are
+     * in the new address book and not in the old address book are greater than or equal to the old next node id and
+     * less than the new next node id.
+     *
+     * @param newNextNodeId  the new next node id
+     * @param oldNextNodeId  the old next node id
+     * @param oldAddressBook the old address book
+     * @param newAddressBook the new address book
+     * @throws IllegalStateException if the new next node id is less than or equal to the old next node id, or if there
+     *                               are any new nodes in the new address book that are less than the old next node id
+     *                               or greater than or equal to the new node id.
+     */
+    public static void validateNewAddressBook(
+            @NonNull final NodeId newNextNodeId,
+            @NonNull final NodeId oldNextNodeId,
+            @NonNull final AddressBook oldAddressBook,
+            @NonNull final AddressBook newAddressBook) {
+        Objects.requireNonNull(newNextNodeId, "The newNextNodeId must not be null.");
+        Objects.requireNonNull(oldNextNodeId, "The oldNextNodeId must not be null.");
+        Objects.requireNonNull(oldAddressBook, "The oldAddressBook must not be null.");
+        Objects.requireNonNull(newAddressBook, "The newAddressBook must not be null.");
+
+        if (newNextNodeId.compareTo(oldNextNodeId) < 0) {
+            throw new IllegalStateException("The newNextNodeId must be greater than or equal to the oldNextNodeId.");
+        }
+
+        final int oldSize = oldAddressBook.getSize();
+        final int newSize = newAddressBook.getSize();
+
+        // Verify that the old next node id is greater than the highest node id in the old address book.
+        final NodeId oldLastNodeId = oldSize > 0 ? oldAddressBook.getNodeId(oldSize - 1) : null;
```

Comment:
Is size 0-based?  If it is this should probably be `oldSize >= 0`.

Reply:
getSize is 0 based.  Note that there is a retrieval with oldSize -1.   If oldSize is greater than 0 then we can access the last address.  If oldSize is 0, then there is no last address to access and the tri-condition returns null in that case. I changed the statement to look like this: 
```
final NodeId oldLastNodeId = (oldSize == 0 ? null : oldAddressBook.getNodeId(oldSize - 1));
```